src/main/java/spark/servlet/SparkServlet.java

```diff
@@ -0,0 +1,273 @@
+
+        // handle static resources
+        boolean consumed = StaticFilesConfiguration.servletInstance.consume(req, resp);
+
+        if (consumed) {
+            return;
+        }
+
+        matcherFilter.doFilter(requestWrapper, resp, null);
+    }
+
+    private static String getConfigPath(String filterMappingPattern, ServletConfig config) {
+        String result = Optional.ofNullable(filterMappingPattern)
+                .orElse(config.getInitParameter(FILTER_MAPPING_PARAM));
+        if (result == null || result.equals(SLASH_WILDCARD)) {
+            return "";
+        } else if (!result.startsWith(SLASH) || !result.endsWith(SLASH_WILDCARD)) {
+            throw new RuntimeException(String.format(
+                    "The %s must start with '/' and end with '/*'. Instead it is: %s",
+                    FILTER_MAPPING_PARAM,
+                    result));
+        }
+        return result.substring(1, result.length() - 1);
+    }
+
+    private static String getRelativePath(HttpServletRequest request, String filterPath) {
+        String path = request.getRequestURI()
+                .substring(request.getContextPath()
+                        .length());
+
+        if (path.length() > 0) {
+            path = path.substring(1);
+        }
+
+        if (filterPath.equals(path + SLASH)) {
+            path += SLASH;
+        }
+
+        if (path.startsWith(filterPath)) {
+            path = path.substring(filterPath.length());
+        }
+
+        if (!path.startsWith(SLASH)) {
+            path = SLASH + path;
+        }
+
+        try {
+            path = URLDecoder.decode(path, "UTF-8");
+        } catch (UnsupportedEncodingException ignore) {
+            // this can't really ever happen
+        }
+
+        LOGGER.debug("Relative path = {}", path);
+        return path;
+    }
+
+    private void populateWrapperSupplier(ServletConfig config) {
+        // Do not override an injected supplier through initialization
+        if (requestSupplier != null) {
+            return;
+        }
+
+        String wrapperSupplierName = config.getInitParameter(WRAPPER_SUPPLIER_PARAM_NAME);
+
+        if (StringUtils.isNotBlank(wrapperSupplierName)) {
+            try {
+                Class<?> wrapperClass = Class.forName(wrapperSupplierName);
+                if (BiFunction.class.isAssignableFrom(wrapperClass)) {
+                    requestSupplier =
+                            (BiFunction<HttpServletRequest, String, HttpServletRequestWrapper>) wrapperClass.newInstance();
+                }
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                LOGGER.debug(
+                        "Error converting {} to BiFunction<HttpServletRequest, String, HttpServletRequestWrapper>; "
+                                + "falling back to default",
+                        wrapperSupplierName,
+                        e);
+            }
+
+        }
+
```

Comment:
My feeling is that it is safe to do so, as missing configuration is not something that would randomly occur at runtime but be consistently missing/wrong during the development cycle. However, I could as easily make the inverse argument, that since it can be resolved by fixing the servlet configuration early, a hard fail here would force that to be done. I can go either way.